core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -382,59 +349,140 @@ func (l *logPollerWrapper) getCurrentCoordinators(ctx context.Context) (common.A
 		Context: ctx,
 	}, donId)
 	if err != nil {
-		return activeCoordinator, l.proposedCoordinator, nil
+		return activeCoordinatorAddress, l.proposedCoordinator.Address(), nil
+	}
+
+	return activeCoordinatorAddress, proposedCoordinator, nil
+}
+
+func (l *logPollerWrapper) getTypeAndVersion(contractAddress common.Address) (string, error) {
+	if contractAddress == (common.Address{}) {
+		l.lggr.Debug("LogPollerWrapper: cannot get typeAndVersion from an unset address")
+		return "", nil
+	}
+
+	contract, err := type_and_version.NewTypeAndVersionInterface(contractAddress, l.client)
+	if err != nil {
+		l.lggr.Error("LogPollerWrapper: could not initialize contract with typeAndVersion interface", contractAddress)
+		return "", err
+	}
+
+	typeAndVersion, err := contract.TypeAndVersion(&bind.CallOpts{})
+	if err != nil {
+		l.lggr.Error("LogPollerWrapper: could not get typeAndVersion from contract ", contractAddress)
+		return "", err
 	}
 
-	return activeCoordinator, proposedCoordinator, nil
+	return typeAndVersion, nil
 }
 
-func (l *logPollerWrapper) handleRouteUpdate(activeCoordinator common.Address, proposedCoordinator common.Address) {
+func (l *logPollerWrapper) handleRouteUpdate(activeCoordinatorAddress common.Address, proposedCoordinatorAddress common.Address) {
 	l.mu.Lock()
 	defer l.mu.Unlock()
 
-	if activeCoordinator == (common.Address{}) {
+	if activeCoordinatorAddress == (common.Address{}) {
 		l.lggr.Error("LogPollerWrapper: cannot update activeCoordinator to zero address")
 		return
 	}
 
-	if activeCoordinator == l.activeCoordinator && proposedCoordinator == l.proposedCoordinator {
+	if (l.activeCoordinator != nil && l.activeCoordinator.Address() == activeCoordinatorAddress) &&
+		(l.proposedCoordinator != nil && l.proposedCoordinator.Address() == proposedCoordinatorAddress) {
```

Comment:
If proposedCoordinator == nil  we never get inside this block but often we should. @agparadiso

Reply:
ok changed to:
```
if (l.activeCoordinator != nil && l.activeCoordinator.Address() == activeCoordinatorAddress) &&
		(l.proposedCoordinator == nil || l.proposedCoordinator.Address() == proposedCoordinatorAddress) {
		l.lggr.Debug("LogPollerWrapper: no changes to routes")
		return
	}
```
that would contemplate both scenarios